.github/workflows/build_deploy_rcb_be.yml

```diff
@@ -0,0 +1,102 @@
+name: Build & Deploy RecipeBox Backend
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+  pull_request:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+  workflow_dispatch: 
+    inputs:
+      run_deploy:
+        type: boolean
+        default: true
+        description: Should the backend app be deployed?
+        required: true
+
+# Azure login
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download and setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Download and install maven
+        run: sudo apt -qq update & sudo apt -qq install maven -y
+
+      - name: Package server app using maven
+        env:
+          DB_URL: ${{ vars.DB_URL }}
+          DB_USERNAME: ${{ vars.DB_USERNAME }}
+          DB_PASSWORD: ${{ secrets.DB_PASSWORD }}
+        # run: mvn package -B -q -DskipTests=true # TODO: remove skipping tests
+        run: mvn package -B -q
```

Comment:
Cred ca e ok, dar probabil nu asa trb, vreau sa adaugi la comanda de mvn package argumente cu variabilele, cauta in documentatia de la mvn cum poti da variabile la comanda

Reply:
O sa schimbam aici cand e gata codul